pkg/network/protocols/events/consumer_test.go

```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
 
 	require.Equalf(t, int(consumer.invalidBatchCount.Get()), 1, "invalidBatchCount should be greater than 0")
 }
+
+func BenchmarkConsumer(b *testing.B) {
+	kversion, err := kernel.HostVersion()
+	require.NoError(b, err)
+	if minVersion := kernel.VersionCode(4, 14, 0); kversion < minVersion {
+		b.Skipf("package not supported by kernels < %s", minVersion)
+	}
+
+	const numEvents = 1000
+	c := config.New()
+	program, err := newEBPFProgram(c)
+	require.NoError(b, err)
+
+	consumer, err := NewConsumer("test", program, func(_ []uint64) {})
+	require.NoError(b, err)
+	generator := newEventGenerator(program, b)
+	require.NoError(b, program.Start())
+
+	for j := 0; j < numEvents; j++ {
+		require.NoError(b, generator.Generate(uint64(j)))
+	}
+	generator.Stop()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
what is purpose of this sleep? do you have any indicatior on how long to wait, not just an arbitrary number?